examples/vector_databases/Vector_db_introduction.ipynb

```diff
@@ -0,0 +1,1031 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'dimension': 1536,\n",
+       " 'index_fullness': 0.2,\n",
+       " 'namespaces': {'content': {'vector_count': 50000},\n",
+       "                'title': {'vector_count': 50000}},\n",
+       " 'total_vector_count': 100000}"
+      ]
+     },
+     "execution_count": 102,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Check index size for each namespace\n",
+    "index.describe_index_stats()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2da40a69",
+   "metadata": {},
+   "source": [
+    "### Search data\n",
+    "\n",
+    "Now we'll enter some dummy searches and check we get decent results back"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 103,
+   "id": "d701b3c7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# First we'll create dictionaries mapping vector IDs to their outputs so we can retrieve the text for our search results\n",
+    "titles_mapped = dict(zip(article_df.vector_id,article_df.title))\n",
+    "content_mapped = dict(zip(article_df.vector_id,article_df.text))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 104,
+   "id": "3c8c2aa1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def query_article(query, namespace, top_k=5):\n",
+    "    '''Queries an article using its title in the specified\n",
+    "     namespace and prints results.'''\n",
+    "\n",
+    "    # Create vector embeddings based on the title column\n",
+    "    embedded_query = openai.Embedding.create(\n",
+    "                                            input=query,\n",
+    "                                            model=MODEL,\n",
+    "                                            )[\"data\"][0]['embedding']\n",
+    "\n",
+    "    # Query namespace passed as parameter using title vector\n",
+    "    query_result = index.query(embedded_query, \n",
+    "                               namespace=namespace, \n",
+    "                               top_k=top_k)\n",
+    "\n",
+    "    # Print query results \n",
+    "    print(f'\\nMost similar results querying {query} in \"{namespace}\" namespace:\\n')\n",
+    "    if not query_result.matches:\n",
+    "        print('no query result')\n",
+    "    \n",
+    "    matches = query_result.matches\n",
+    "    ids = [res.id for res in matches]\n",
+    "    scores = [res.score for res in matches]\n",
+    "    df = pd.DataFrame({'id':ids, \n",
+    "                       'score':scores,\n",
+    "                       'title': [titles_mapped[_id] for _id in ids],\n",
+    "                       'content': [content_mapped[_id] for _id in ids],\n",
+    "                       })\n",
+    "    \n",
+    "    counter = 0\n",
+    "    for k,v in df.iterrows():\n",
+    "        counter += 1\n",
+    "        print(f'Result {counter} with a score of {v.score} is {v.title}')\n",
```

Comment:
Starting each line with Result #1 #2 etc feels a bit low-information density. Could also do something like "Battle of Bannockburn (score = 0.869)" so that the results are immediately readable without needing to read all the way across the line. Your call.

Reply:
Updated and made consistent across the two vector DBs